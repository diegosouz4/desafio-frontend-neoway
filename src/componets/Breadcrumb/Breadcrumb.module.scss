.breadCrumb {
  text-align: center;
  a {
    color: var(--color-white-100);
    font-size: 0.875rem;
    font-weight: 400;
    &:hover,
    &:focus {
      text-decoration: underline;
    }

    &::after {
      content: "/";
      color: var(--color-white-100);
      display: inline-block;
      margin: 0 0.3125rem;
    }
  }

  span {
    color: var(--color-white);
    font-weight: 700;
    border-bottom: 2px solid var(--color-main);
  }
}
