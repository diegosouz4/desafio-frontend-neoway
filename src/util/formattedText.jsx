export default function formattedText(text, limit = 50) {
  if (!text || text.length < limit) return text;
  let formattedText = text.substring(0, limit);
  formattedText = `${formattedText}...`;
  return formattedText;
}
