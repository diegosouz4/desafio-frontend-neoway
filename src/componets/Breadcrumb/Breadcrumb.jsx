import { Link } from "react-router-dom";
import style from "./Breadcrumb.module.scss";
import PropTypes from "prop-types";
import formattedText from "../../util/formattedText";

export default function Breadcrumb({ title, className }) {
  const formattedTitle = formattedText(title, 20);

  return (
    <div className={`${className} ${style.breadCrumb}`}>
      <Link to="/">home</Link>
      <span>{formattedTitle}</span>
    </div>
  );
}

Breadcrumb.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
};
