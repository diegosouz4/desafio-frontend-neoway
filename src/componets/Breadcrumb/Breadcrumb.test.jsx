import { render, screen } from "@testing-library/react";
import Breadcrumb from "./Breadcrumb";
import { BrowserRouter } from "react-router-dom";
import formattedText from "../../util/formattedText";
import "@testing-library/jest-dom";

const mockInfo = {
  title: "Example Title",
};

describe("<Breadcrumb/>", () => {
  it("Should be rendering", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Breadcrumb title={mockInfo.title} />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should showing the home link", () => {
    render(
      <BrowserRouter>
        <Breadcrumb title={mockInfo.title} />
      </BrowserRouter>
    );

    const homeLink = screen.getByText("home");

    expect(homeLink).toBeInTheDocument();
    expect(homeLink.getAttribute("href")).toBe("/");
  });

  it("should showing the title passed as props", () => {
    render(
      <BrowserRouter>
        <Breadcrumb title={mockInfo.title} />
      </BrowserRouter>
    );

    const title = screen.getByText(mockInfo.title);
    expect(title).toBeInTheDocument();
  });

  it("should show the custom class passed as props", () => {
    const className = "jest-class";

    render(
      <BrowserRouter>
        <Breadcrumb title={mockInfo.title} className={className} />
      </BrowserRouter>
    );

    const component = screen.getByText(mockInfo.title).parentElement;
    expect(component).toHaveClass(className);
  });

  it("should title not be formatted", () => {
    const text = "Short text";
    const result = formattedText(text);

    render(
      <BrowserRouter>
        <Breadcrumb title={result} />
      </BrowserRouter>
    );

    const title = screen.getByText(text);
    expect(title).toBeInTheDocument();
  });

  it("should title be formatted", () => {
    const text = "This is a long text that needs to be formatted";
    const result = formattedText(text);

    render(
      <BrowserRouter>
        <Breadcrumb title={result} />
      </BrowserRouter>
    );

    // screen.debug();

    const title = screen.getByText("This is a long text ...");
    expect(title).toBeInTheDocument();
  });
});
