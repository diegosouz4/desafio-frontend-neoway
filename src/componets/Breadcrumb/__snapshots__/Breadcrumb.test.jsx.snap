// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb/> Should be rendering 1`] = `
<DocumentFragment>
  <div
    class="undefined undefined"
  >
    <a
      href="/"
    >
      home
    </a>
    <span>
      Example Title
    </span>
  </div>
</DocumentFragment>
`;
